pom.xml

```diff
@@ -366,6 +367,17 @@ limitations under the License.
                     <artifactId>maven-scm-publish-plugin</artifactId>
                     <version>${scm-publish-plugin.version}</version>
                 </plugin>
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-compiler-plugin</artifactId>
+                    <version>${compiler-plugin.version}</version>
+                    <configuration>
+                        <compilerArgs>
+                            <arg>-verbose</arg>
+                            <arg>-Xlint:all,-options,-path</arg>
+                        </compilerArgs>
+                    </configuration>
+                </plugin>
```

Comment:
I should have written down why I added this, but I think it is because with these options the Java compiler is way more verbose about unchecked warnings (maybe it upgrades those unchecked to actual errors). I am in favor of leaving it this way.

Reply:
Hm. Not generally a fan of increased verbosity just for the sake of verbosity (you don't run maven in debug mode all the time, do you? :smile:) but I do like strict compilers. Did this reveal any real issues for you, though? I feel like most of the "bad stuff" should already be detected by checkstyle/pmd/findbugs (maybe tweak some configs there?).

I know that some of the generated code (AutomatonBuilders) raises some unchecked warnings. If this only prints more details about the issues, it's fine. But the code should still compile without an error.

Reply:
Ok, I found out what this change does. It will add the line number to e.g. forgotten `<>`. I.e. without the options you get:

    [INFO] /home/jeroen/Documents/gitprojects/automatalib/serialization/fsm/src/main/java/net/automatalib/serialization/fsm/parser/FSM2DFAParser.java: /home/jeroen/Documents/gitprojects/automatalib/serialization/fsm/src/main/java/net/automatalib/serialization/fsm/parser/FSM2DFAParser.java uses unchecked or unsafe operations.
    [INFO] /home/jeroen/Documents/gitprojects/automatalib/serialization/fsm/src/main/java/net/automatalib/serialization/fsm/parser/FSM2DFAParser.java: Recompile with -Xlint:unchecked for details.

With the options you get:

    [WARNING] /home/jeroen/Documents/gitprojects/automatalib/serialization/fsm/src/main/java/net/automatalib/serialization/fsm/parser/FSM2DFAParser.java:[313,16] unchecked call to FSM2DFAParser(java.io.Reader,java.util.function.Function<java.lang.String,I>,java.lang.String,java.lang.String) as a member of the raw type net.automatalib.serialization.fsm.parser.FSM2DFAParser
    [WARNING] /home/jeroen/Documents/gitprojects/automatalib/serialization/fsm/src/main/java/net/automatalib/serialization/fsm/parser/FSM2DFAParser.java:[313,110] unchecked conversion
      required: net.automatalib.automata.fsa.impl.compact.CompactDFA<I>
      found:    net.automatalib.automata.fsa.impl.compact.CompactDFA

So it gives you 
1. the line number, and
1. a suggested fix, and
1. a [WARNING] instead of an [INFO]

So this will not reject the issues you mentioned in the builder classes, instead it proves more information about the unchecked conversions. So, shall we keep it?

Reply:
Sure, looks nice.